File: python/structural/facade/publisher/html_converter.py
def plain_to_html(content: str) -> str:
    lines = content.strip().split("\n")
    html_lines = []
    in_list = False

    for line in lines:
        line = line.strip()
        
        if line.startswith("## "):
            html_lines.append(f"<h2>{line[3:]}</h2>")
        elif line.startswith("# "):
            html_lines.append(f"<h1>{line[2:]}</h1>")
        elif line.startswith("- "):
            if not in_list:
                html_lines.append("<ul>")
                in_list = True
            html_lines.append(f"<li>{line[2:]}</li>")
        else:
            if in_list:
                html_lines.append("</ul>")
                in_list = False
            html_lines.append(f"<p>{line}</p>")

    if in_list:
        html_lines.append("</ul>")

    return "\n".join(html_lines)

File: typescript/src/structural/facade/publisher/HtmlConverter.ts
export function plainToHtml(content: string) : string{
    const lines = content.trim().split("\n")
    const htmlLines = []
    let inList = false

    for (let line in lines){
        line = line.trim()

        if (line.startsWith("## ")){
            htmlLines.push(`<h2>${line.slice(3)}</h2>`)
        }
        else if(line.startsWith("# ")){
            htmlLines.push(`<h1>${line.slice(2)}</h2>`)
        }
        else if(line.startsWith("- ")){
            if (!inList){
                htmlLines.push("<ul>")
                inList = true
            }
            htmlLines.push(`<li>${line.slice(3)}</li>`)
        }
        else{
            if (inList){
                htmlLines.push("</ul>")
                inList = false
            }
            htmlLines.push(`<p>${line}</p>`)
        }

    }

    if (inList){
        htmlLines.push("</ul>")
    }
    
    return htmlLines.join("\n")
}